### PR TITLE
tag fix - improvement

### PR DESCRIPTION
**Pull request by phweyland**
- allows pipe character in create tag
- allows entry tag(s) creation without image selected
- makes visible and selects the newly created tag in dictionary view
- removes selection indicator in attached view when only one image (selected or hovered)
- fixes wrong indicators when attach tag without selected image
- enum for selection states

Covers some of @GwenaelQ issues / suggestions: #4810, #4812, #4813.

### CODE REVIEW

File: src/common/tags.c [hunk 1]
```diff
@@ -510,44 +510,44 @@ void dt_tag_set_tags(const GList *tags, const GList *img, const gboolean ignore_
...
   // undo only undoes the tags attachments. it doesn't remove created tags.
   gchar **tokens = g_strsplit(tags, ",", 0);
   if(tokens)
   {
+    // tag(s) creation
+    GList *tagl = NULL;
+    gchar **entry = tokens;
+    while(*entry)
+    {
+      // remove leading and trailing spaces
```
Comment (TurboGit) [inline src/common/tags.c:523]: Maybe use g_strstrip()

File: src/libs/tagging.c [hunk 2]
```diff
@@ -221,7 +221,64 @@ static void _sort_dictionary_list(dt_lib_module_t *self, gboolean force)
...
+  return found;
+}
+
+// make the tag visible on view
+static void _show_tag_on_view(GtkTreeView *view, const char *tagname)
+{
+  if(tagname)
+  {
+    char *lt = g_strdup(tagname);
+    // remove leading and trailing spaces
```
Comment (TurboGit) [inline src/libs/tagging.c:257]: likewise g_strstrip()

File: src/libs/tagging.c [hunk 3]
```diff
@@ -506,6 +567,7 @@ static gboolean _find_tag_iter_tagid(GtkTreeModel *model, GtkTreeIter *iter, gin
   return FALSE;
 }
 
+#if 0
```
Comment (TurboGit) [inline src/libs/tagging.c:570]: Either the code should be removed or some comments added to why it is commented out.

Comment (TurboGit) [reply src/libs/tagging.c:570]: I'm in favor of removing, it is in Git anyway.

File: src/common/tags.c (hunk 1 above)
Comment (phweyland) [reply src/common/tags.c:523]: Great !

File: src/libs/tagging.c (hunk 3 above)
Comment (phweyland) [reply src/libs/tagging.c:570]: >  it is in Git anyway.

True. I'll remove it.

**Comment by phweyland** [comment]
Review comments integrated. Thanks.

**Comment by TurboGit** [comment]
>     * makes visible and selects the newly created tag in dictionary view

Can you explain this one ? Where should I look for my testing ?

**Comment by phweyland** [comment]
entering aa|bb|cc you should get something like :

image

Entering aa|bb|dd, zz|aa|zz should give something like (open the last tag entered):

image

Should work in list view as well but not in suggestion view (as already attached tags are not shown).
Should work also with create tag and rename path (context menu).

**Comment by TurboGit** [comment]
I don't get the selected text on my side:

image

**Comment by TurboGit** [comment]
I have tested when the selected image as the corresponding tag or not.

**Comment by phweyland** [comment]
> I have tested when the selected image as the corresponding tag or not.

The view you show is a filtered one.(selection has gone for a while).
What do you see just after enter ?

**Comment by TurboGit** [comment]
> What do you see just after enter ?

It is selected as shown on your screenshot. Perfect then.

> * makes visible and selects the newly created tag in dictionary view

I read this as two behaviors "make visible created tag in dictionary view" and "select the newly created tag in dictionary", my bad why reading the simple sentence when one can read something overly complex. Anyway, now I see it all good.... Merging.

Thanks.

**Pull request merged**